src/acceptance-test/java/org/zalando/nakadi/webservice/hila/StreamBatch.java

```diff
@@ -36,30 +40,54 @@ public SubscriptionCursor getCursor() {
         return unmodifiableList(events);
     }
 
+    public StreamMetadata getMetadata() {
+        return metadata;
+    }
+
+    public static StreamBatch singleEventBatch(final String partition, final String offset, final String eventType,
+                                               final Map event, final String metadata) {
+        if (event.isEmpty()) {
+            return new StreamBatch(new SubscriptionCursor(partition, offset, eventType, DUMMY_TOKEN),
+                    ImmutableList.of(), new StreamMetadata(metadata));
+        } else {
+            return new StreamBatch(new SubscriptionCursor(partition, offset, eventType, DUMMY_TOKEN),
+                    ImmutableList.of(event), new StreamMetadata(metadata));
+        }
+    }
+
     public static StreamBatch singleEventBatch(final String partition, final String offset, final String eventType,
                                                final Map event) {
         return new StreamBatch(new SubscriptionCursor(partition, offset, eventType, DUMMY_TOKEN),
-                ImmutableList.of(event));
+                ImmutableList.of(event), null);
     }
 
     @Override
     public boolean equals(final Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
+
         final StreamBatch that = (StreamBatch) o;
-        return cursor.equals(that.cursor) && events.equals(that.events);
+
+        if (!cursor.equals(that.cursor)) return false;
+        if (events != null ? !events.equals(that.events) : that.events != null) return false;
```

Comment:
But I mean the implementation, not the API.
It's implemented in a way that it would be empty if there are no events https://github.com/zalando/nakadi/pull/406/files#diff-f5f9dae0f2568532a30e93bda9f32536L28